paper.tex

```diff
@@ -0,0 +1,223 @@
+\documentclass[format=acmsmall]{acmart}
+\begin{document}
+\title{GPU-friendly Stroke Expansion}
+\author{Raph Levien}
+\author{Arman Uguray}
+\affiliation{
+    \institution{Google}
+    \city{San Francisco}
+    \state{CA}
+    \country{USA}
+}
+
+\begin{abstract}
+    Vector graphics includes both filled and stroked paths as the main primitives. While there have been many proposed techniques for rendering filled paths on GPU, stroked paths have proved more elusive. This paper presents a technique for performing stroke expansion, namely the generation of an outline representing the stroke of the given input path, an operation which is considered global, but which we nonetheless implement using a fully parallel algorithm. Our method relies on several novel techniques, including an encoding of the input suitable for parallel decoding and processing, an Euler spiral based method for computing flattened parallel curves. While we focus on flattened output consisting of polylines, the methods can be adapted to preserving curves in the output as well.
+\end{abstract}
+
+\maketitle
+\thispagestyle{empty}
+\pagestyle{plain}
+
+\section{Introduction}
+
+Rendering of vector graphics documents requires handling both filled and stroked primitives. There is substantial literature on GPU rendering of filled paths, but many fewer published techniques for strokes. It is a more challenging problem, especially for parallel computation, because path segments cannot be processed independently of each other; the \emph{joins} between adjacent path segments depend on context. In addition, the path topology affects the rendered result, in particular whether subpaths are open or closed. In the former case, the endpoints are rendered with \emph{caps}. Joins and caps can have multiple styles, including different styles within the same document.
```

Comment:
I think the sentence explaining the challenging nature of stroking should mention curve offsetting as one of the challenges. Some suggestions (feel free to rephrase and/or reject them at your discretion):

```suggestion
Rendering of vector graphics documents requires handling both filled and stroked primitives. The most common type of primitive is a \emph{path} composed of one or more sequences of connected Bézier curve segments. There is substantial literature on GPU rendering of filled paths, but many fewer published techniques for strokes. Rendering of stroked primitives is a more challenging problem: i) it is necessary to compute the parallel curves outlining the stroked shape at a fixed distance from the source curve, a computationally expensive procedure that must account for discontinuities; ii) path segments cannot be processed independently of each other, as adjacent segments must be \emph{joined} without breaking the smoothness of the contour. In addition, the path topology affects the rendered result, in particular whether subpaths are open or closed. In the former case, the endpoints are rendered with \emph{caps}. Joins and caps can have multiple styles, including different styles within the same document.
```